openpgp/s2k/s2k_config.go

```diff
@@ -35,6 +36,11 @@ type Config struct {
 	// use a value that is at least 65536. See RFC 4880 Section
 	// 3.7.1.3.
 	S2KCount int
+	// When true, allows the S2KMode to be s2k.SaltedS2K.
+	// This is required to ensure the application is aware that a weaker
+	// S2K is being selected, and is only being used when the key has
+	// been randomly generated or a secure S2K has already been applied.
+	HighEntropyKey bool
```

Comment:
I think there's some small risk that someone might only look at this name and say "do I want a high-entropy key? well yes, of course!" and think that true is better than false. How about
```suggestion
	PassphraseIsHighEntropy bool
```
, which while verbose, should make it super clear what it means.